src/pages/ArticlePage.jsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { fetchArticleById } from "../api";
+
+function ArticlePage() {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    fetchArticleById(article_id)
+      .then((fetchedArticle) => {
+        setArticle(fetchedArticle);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        console.error(err);
+        setIsLoading(false);
+      });
+  }, [article_id]);
+
+  if (isLoading) return <p>Loading article...</p>;
+  if (!article) return <p>Article not found</p>;
```

Comment:
I love that you've included loading and error handling for the users 🏆 

Consider refactoring when you get time to use a state for the error, so that in your `catch` you could `setError(true)` instead of console logging. This is just a bit safer than using `!article` and the state can hold data of the error returned if you want it to